src/seamless_communication/denoise/demucs.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# MIT_LICENSE file in the root directory of this source tree.
+
+from pathlib import Path
+import subprocess as sp
+import tempfile
+from typing import Union
+from torch import Tensor
+import torchaudio
+from fairseq2.memory import MemoryBlock
+from dataclasses import dataclass
+from typing import Optional
+
+SAMPLING_RATE = 16000
+
+@dataclass
+class DenoisingConfig:
+    def __init__(
+            self,
+            filter_width: int = 3,
+            model="htdemucs", 
+            sample_rate=SAMPLING_RATE,
+            two_stems=None,
+            float32=False,
+            int24=False):
+        self.filter_width = filter_width
+        self.model = model
+        self.sample_rate = sample_rate
+        self.two_stems = two_stems
+        self.float32 = float32
+        self.int24 = int24
+
+class Demucs():
+    def __init__(
+            self, 
+            denoise_config: Optional[DenoisingConfig]):
+        self.denoise_config = denoise_config
+
+    def run_command_with_temp_file(self, cmd):
+        with tempfile.NamedTemporaryFile(mode='w+', delete=False) as temp:
+            result = sp.run(cmd, stdout=temp, stderr=temp, text=True)
+            # If there was an error, print the content of the file
+            if result.returncode != 0:
+                temp.seek(0)
+                print(temp.read())
+
+    def denoise(self, audio: Union[str, Tensor]):
+
+        if self.denoise_config is None:
+          self.denoise_config = DenoisingConfig()
+
+        if isinstance(audio, Tensor):
+            with tempfile.NamedTemporaryFile(suffix=".wav", delete=False) as temp_wav:
+                torchaudio.save(temp_wav.name, audio, self.denoise_config.sample_rate)
```

Comment:
nit: let's make sure to clean all temp files

Reply:
Fixed